Issue2948 motor driver update

JayHuLBL opened the pull request:
This is a draft pull request.

@ViswanathanGanesh01  Please see the comments below:
- [x] Missing comments for instances in the classes, in folder `Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.InductionMotors.BaseClasses`.
- [x] The base classes do not have documentation and revision note.
- [x] There still some mixed use of the equation and connections in the equation section. For example, in class `Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.InductionMotors.SquirrelCage`.
- [x] Run unit test with Optimica, we can still see the following warnings. It shows that some self-connectings. And also, `*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.InductionMotors.Examples.SquirrelCageDrive: Access to component v not recommended, it is not present in constraining type of declaration 'replaceable Buildings.Electrical.Interfaces.BaseTerminal terminal'`.
```
(py3_11_3) jianjunhu@ubuntu:~/GitFolder/modelica-buildings/Buildings$ ../bin/runUnitTests.py -s Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive -t optimica
Regression tests are only run for the following package:
  Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive
Number of models   : 2658
          blocks   : 635
          functions: 222
Using 4 of 4 processors to run unit tests for optimica.
Generated 8 regression tests.

*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.InductionMotors.Examples.SquirrelCageDrive: Access to component v not recommended, it is not present in constraining type of declaration 'replaceable Buildings.Electrical.Interfaces.BaseTerminal terminal'
*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.InductionMotors.Examples.SquirrelCageDrive: Access to component i not recommended, it is not present in constraining type of declaration 'replaceable Buildings.Electrical.Interfaces.BaseTerminal terminal'
*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.InductionMotors.Examples.SquirrelCageDrive: Access to component theta not recommended, it is not present in constraining type of declaration 'replaceable Buildings.Electrical.Interfaces.BaseTerminal terminal'
*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.InductionMotors.Examples.SquirrelCageDrive: Ignored connection from connector to itself
*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.InductionMotors.Examples.SquirrelCageDrive: Ignored connection from connector to itself
*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.Coupled.Examples.Chiller: Access to component v not recommended, it is not present in constraining type of declaration 'replaceable Buildings.Electrical.Interfaces.BaseTerminal terminal'
*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.Coupled.Examples.Chiller: Access to component i not recommended, it is not present in constraining type of declaration 'replaceable Buildings.Electrical.Interfaces.BaseTerminal terminal'
*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.Coupled.Examples.Chiller: Access to component theta not recommended, it is not present in constraining type of declaration 'replaceable Buildings.Electrical.Interfaces.BaseTerminal terminal'
*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.Coupled.Examples.Chiller: Ignored connection from connector to itself
*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.Coupled.Examples.Chiller: Ignored connection from connector to itself
*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.Coupled.Examples.Pump: Access to component v not recommended, it is not present in constraining type of declaration 'replaceable Buildings.Electrical.Interfaces.BaseTerminal terminal'
*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.Coupled.Examples.Pump: Access to component i not recommended, it is not present in constraining type of declaration 'replaceable Buildings.Electrical.Interfaces.BaseTerminal terminal'
*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.Coupled.Examples.Pump: Access to component theta not recommended, it is not present in constraining type of declaration 'replaceable Buildings.Electrical.Interfaces.BaseTerminal terminal'
*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.Coupled.Examples.Pump: Ignored connection from connector to itself
*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.Coupled.Examples.Pump: Ignored connection from connector to itself
*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.Coupled.Examples.Pump: Ignored connection from connector to itself
*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.InductionMotors.Examples.SquirrelCageStartUp: Access to component v not recommended, it is not present in constraining type of declaration 'replaceable Buildings.Electrical.Interfaces.BaseTerminal terminal'
*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.InductionMotors.Examples.SquirrelCageStartUp: Access to component i not recommended, it is not present in constraining type of declaration 'replaceable Buildings.Electrical.Interfaces.BaseTerminal terminal'
*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.InductionMotors.Examples.SquirrelCageStartUp: Access to component theta not recommended, it is not present in constraining type of declaration 'replaceable Buildings.Electrical.Interfaces.BaseTerminal terminal'
*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.InductionMotors.Examples.SquirrelCageStartUp: Ignored connection from connector to itself
*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.InductionMotors.Examples.SquirrelCageStartUp: Ignored connection from connector to itself
*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.Coupled.Examples.HeatPump: Access to component v not recommended, it is not present in constraining type of declaration 'replaceable Buildings.Electrical.Interfaces.BaseTerminal terminal'
*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.Coupled.Examples.HeatPump: Access to component i not recommended, it is not present in constraining type of declaration 'replaceable Buildings.Electrical.Interfaces.BaseTerminal terminal'
*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.Coupled.Examples.HeatPump: Access to component theta not recommended, it is not present in constraining type of declaration 'replaceable Buildings.Electrical.Interfaces.BaseTerminal terminal'
*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.Coupled.Examples.HeatPump: Ignored connection from connector to itself
*** Warning: Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.Coupled.Examples.HeatPump: Ignored connection from connector to itself
```

- [x] Run `../bin/verifyFiles.py `, we have the following errors. It means the class does not documentation.
```
jianjunhu@ubuntu:~/GitFolder/modelica-buildings/Buildings$ ../bin/verifyFiles.py 
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/ThermoFluid/BaseClasses/package.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/ThermoFluid/BaseClasses/Power.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/RotorCurrent_d.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/CurrentBlock.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/CurrentCalculationQ.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/TorqueBlock.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/CurrentCalculationQ_VFD.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/RotorCurrent_q.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/VoltageConversion.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/RMS_Voltage.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/FrequencyConversion.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/SimVFD.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/StatorCurrent_d.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/SpeedBlock.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/CurrentCalculationD.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/StatorCurrent_q.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/CurrentCalculationD_VFD.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/package.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/Data/IM_10HP_400V_50Hz.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/Data/Generic.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/Data/IM_100HP_460V_60Hz.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/Data/IM_20HP_460V_60Hz.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/Data/IM_20HP_400V_50Hz.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/Data/IM_100HP_400V_50Hz.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/Data/IM_50HP_460V_60Hz.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/Data/IM_200HP_460V_60Hz.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/Data/IM_5HP_460V_60Hz.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/Data/IM_200HP_400V_50Hz.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/Data/IM_5HP_400V_50Hz.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/Data/IM_10HP_460V_60Hz.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/Data/IM_50HP_400V_50Hz.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/Data/IM_150HP_460V_60Hz.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/Data/IM_150HP_400V_50Hz.mo' does not contain required string 'documentation'.
*** Error:  File 'Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/Data/package.mo' does not contain required string 'documentation'.
*** Terminating due to found errors in examined files.
```

- [x] Classes (except validation and example models) should have `defaultComponentName`. See for example https://github.com/lbl-srg/modelica-buildings/blob/ca1e43ffb7b0f8c3ac4f7376ff0542472544a966/Buildings/Controls/OBC/ASHRAE/G36/AHUs/SingleZone/VAV/Controller.mo#L1265
- [x] For all the example and validation models, make sure the .mos script settings are consistent with the one in .mo file. For example, in class `Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.Coupled.Examples.Pump`, it has `annotation (experiment(Tolerance=1e-6,StopTime=600, StartTime=400),`. But the corresponding .mos script has `tolerance=1e-6,startTime=0, stopTime=200, `.

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Create package.order

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Create package.oder

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Delete Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/ThermoFluid/package.oder

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Delete Buildings/Resources/Scripts/Dymola/Electrical/AC/ThreePhasesBalanced/Loads/Examples/MotorDrive/InductionMotors/BaseClasses directory

ViswanathanGanesh01 committed:
Delete Buildings/Resources/Scripts/Dymola/Electrical/AC/ThreePhasesBalanced/Loads/Examples/MotorDrive/InductionMotors/Examples directory

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Delete Buildings/Resources/Scripts/Dymola/Electrical/AC/ThreePhasesBalanced/Loads/Examples/MotorDrive directory

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Delete Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive directory

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Delete Buildings/Resources/Scripts/Dymola/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/Coupled/Examples directory

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Delete Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/MotorModel1.mo

Duplicate file.

ViswanathanGanesh01 committed:
Update package.order

Removed MotorModel1

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Delete Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/computeElectricalParameters.bak-mo

ViswanathanGanesh01 committed:
Merge branch 'lbl-srg:master' into issue2948-MotorDrive

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Delete Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/ThermoFluid/BaseClasses.mo

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

JayHuLBL committed:
Merge pull request #4103 from ViswanathanGanesh01/issue2948-MotorDrive

Updates to the Motor Drive Package

JayHuLBL committed:
added reference, openmoldeica

JayHuLBL committed:
removed BOM

JayHuLBL committed:
corrected wrong using of scripts

JayHuLBL committed:
corrected package orders

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Merge branch 'issue2948_MotorDriver_update' into issue2948-MotorDrive

JayHuLBL committed:
Merge pull request #4286 from ViswanathanGanesh01/issue2948-MotorDrive

Issue2948 motor drive

JayHuLBL committed:
updated reference results

JayHuLBL committed:
refactoring models

JayHuLBL committed:
added show_t

ViswanathanGanesh01 commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/ThermoFluid/BaseClasses/Power.mo:
```diff
@@ -0,0 +1,19 @@
+
+input Real tau "Torque";
+input Real spe "Rotational Speed";
+input Real upper "Upper Limit for smooth function";
```
@JayHuLBL , the `upper` and `lower` are just intended as placeholders to smooth the variable `spe` because this depicts the machines mechanical torque that needs to be smooth.

JayHuLBL committed:
Merge branch 'master' into issue2948_MotorDriver_update

JayHuLBL committed:
corrected expertimental setup

JayHuLBL committed:
improving graphics

JayHuLBL committed:
created based model for the motor induction models

JayHuLBL committed:
refactoring the coupled models

JayHuLBL committed:
improving coupled models [ci skip]

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/Coupled/Pump.mo:
```diff
@@ -0,0 +1,228 @@
+  parameter Modelica.Units.SI.Time tau=1
+    "Time constant of fluid volume for nominal flow, used if energy or mass balance is dynamic"
+    annotation (Dialog(tab="Dynamics", group="Conservation equations"));
+  parameter Boolean use_riseTime=true
```
Will the rise time affect the motor rotation?

JayHuLBL replied:
How does the rise time and motor rotation affect each other? Is it possible to add the clarification into the documentation? User would like to know how to set it and what will happen if the rise time changes.

ViswanathanGanesh01 commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/Coupled/Pump.mo:
```diff
@@ -0,0 +1,228 @@
+
+  //Motor parameters
+  replaceable parameter Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.InductionMotors.Data.Generic
+    per1 constrainedby Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.InductionMotors.Data.Generic
```
No Dr. Hu, per and per1 are the same.

JayHuLBL replied:
Could you please be more specific saying that "`per` and `per1` are the same"? If they are indeed the same, why do we need to have both of them.
It would be better to clarify the two data set in the documentation.

ViswanathanGanesh01 replied:
Sure Dr. Hu, I will address the above comments in PR #4292

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/Coupled/Pump.mo:
```diff
@@ -0,0 +1,228 @@
+    "Mechanical pump with mechanical interface"
+    annotation (Placement(transformation(extent={{-10,-10},{10,10}})));
+
+  Modelica.Blocks.Interfaces.RealInput setPoi "Set point of control target"
```
What is this set point? is it speed (0~1, or the real rotation speed), volume flow rate, mass flow rate? The comment should be clear.

ViswanathanGanesh01 replied:
The set point is the mass flow rate.

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/Coupled/Pump.mo:
```diff
@@ -0,0 +1,228 @@
+          textString="measure_value")}),
+        defaultComponentName="pum",
+    Documentation(info="<html>
+<p>
```
Improve the documentation to include detailed instructions on how to use the model.

JayHuLBL replied:
@ViswanathanGanesh01 could you address this comment in PR #4292?

ViswanathanGanesh01 replied:
Sure Dr. Hu, I will address the above comments in PR #4292

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/CurrentBlock.mo:
```diff
@@ -0,0 +1,27 @@
+within Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.InductionMotors.BaseClasses;
```
For most of the models in this base class package, the comments for each instance are missing.

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/CurrentBlock.mo:
```diff
@@ -0,0 +1,27 @@
+within Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.InductionMotors.BaseClasses;
+model CurrentBlock "Converts DQ-abc"
+  extends Modelica.Blocks.Icons.Block;
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput i_ds
```
add comment

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/CurrentCalculationD.mo:
```diff
@@ -0,0 +1,17 @@
+within Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.InductionMotors.BaseClasses;
+function CurrentCalculationD   "Function to compute current to interface with electrical grid"
+
+  input Real i_ds;
```
add comment

JayHuLBL reviewed (commented):
@ViswanathanGanesh01 Please see inline comments. Could you please address the comments and also update models or the documentation to avoid the questions I asked, by creating a new PR?

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Merge branch 'issue2948_MotorDriver_update' into MBL---Electrical-Package-v1

ViswanathanGanesh01 committed:
Add files via upload

JayHuLBL committed:
Merge pull request #4292 from ViswanathanGanesh01/MBL---Electrical-Package-v1

Motor Drive Package: Updated documentation and inline Comments

JayHuLBL committed:
improved text format

JayHuLBL committed:
modified text

JayHuLBL committed:
added prefix r

JayHuLBL committed:
upgraded node version

JayHuLBL committed:
upgraded node version before installation

JayHuLBL committed:
switch to the desired node version

JayHuLBL committed:
added sudo

JayHuLBL committed:
changed approach to upgrade node

JayHuLBL committed:
changed back node upgrade approach

JayHuLBL committed:
added sudo

JayHuLBL committed:
add npm installation

JayHuLBL committed:
added rehash

JayHuLBL committed:
added command end

JayHuLBL committed:
added apt update

JayHuLBL committed:
moved the node version update to the test script

JayHuLBL committed:
added npm installation

JayHuLBL committed:
updated node js version

JayHuLBL committed:
changed yml file syntax error

JayHuLBL committed:
corrected .yml error

JayHuLBL committed:
added specific node version

JayHuLBL committed:
changed to the latest node version

JayHuLBL committed:
changed to the latest node version

JayHuLBL committed:
Merge branch 'issue4306_addPrefixR' into issue2948_MotorDriver_update

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Improved Documentation and Validation

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Add files via upload

ViswanathanGanesh01 committed:
Added Validation Scripts

ViswanathanGanesh01 committed:
Merge branch 'issue2948_MotorDriver_update' into MBL---Electrical-Package-v1

ViswanathanGanesh01 committed:
Removed Initial Dynamics Override

Removed Override: "pum(energyDynamics=Modelica.Fluid.Types.Dynamics.FixedInitial)"

ViswanathanGanesh01 committed:
Added additional documentation

ViswanathanGanesh01 committed:
Added Data for Validations

ViswanathanGanesh01 committed:
Updated Script for TableonFile

ViswanathanGanesh01 committed:
Delete Buildings/Resources/Scripts/Dymola/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/Validations/MSL.csv

ViswanathanGanesh01 committed:
Delete Buildings/Resources/Scripts/Dymola/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/Validations/Lefosse2021.csv

ViswanathanGanesh01 committed:
Added Unit and quantity for parameters

JayHuLBL committed:
Merge pull request #4358 from ViswanathanGanesh01/MBL---Electrical-Package-v1

Improved Documentation and Added Validation package

JayHuLBL committed:
improving format [ci skip]

JayHuLBL committed:
merged master [ci skip]

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/CurrentCalculationD.mo:
```diff
@@ -0,0 +1,24 @@
+within Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.InductionMotors.BaseClasses;
+function CurrentCalculationD   "Function to compute current to interface with electrical grid"
```
The comment `"Function to compute current to interface with electrical grid"` is used by functions `CurrentCalculationD`, `CurrentCalculationD_VFD`, `CurrentCalculationQ`, and `CurrentCalculationQ_VFD`.

Please change the comments to the user how the function is doing.

Also, in the function name, there is no need to have `Calculation`.

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/PartialSquirrelCage.mo:
```diff
@@ -0,0 +1,113 @@
+  Modelica.Blocks.Sources.RealExpression volPhaAng(final y=theta_s)
+    "Supply voltage phase angle"
+    annotation (Placement(transformation(extent={{-80,70},{-60,90}})));
+  Modelica.Blocks.Continuous.Integrator int
```
missing comment

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/RotorCurrent_d.mo:
```diff
@@ -0,0 +1,65 @@
+    "Derivative of D-axis rotor current"
+    annotation (Placement(transformation(extent={{100,-20},{140,20}})));
+
+initial equation
```
Do we need the `initial equation` section?

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/SimVFD.mo:
```diff
@@ -0,0 +1,109 @@
+    annotation (Placement(transformation(extent={{-64,50},{-44,70}})));
+  Modelica.Blocks.Math.Division VFD_Ratio "VFD ratio"
+    annotation (Placement(transformation(extent={{-20,40},{0,60}})));
+  Modelica.Blocks.Sources.RealExpression realExpression(y=N_s*p/120)
```
missing comment

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/SquirrelCage.mo:
```diff
@@ -0,0 +1,87 @@
+    annotation (Placement(transformation(extent={{-140,-100},{-100,-60}}),
+        iconTransformation(extent={{-140,-100},{-100,-60}})));
+  Buildings.Controls.OBC.CDL.Reals.MultiplyByParameter gai(
+    final k=1/(2*Modelica.Constants.pi))
```
missing comment

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/Validations/DOLStartUp.mo:
```diff
@@ -0,0 +1,82 @@
+    "Voltage source"
+    annotation (Placement(transformation(extent={{-20,20},{0,40}})));
+  Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.InductionMotors.SquirrelCage motDri
+    annotation (Placement(transformation(extent={{-18,-20},{2,0}})));
```
missing comment

JayHuLBL commented:
@ViswanathanGanesh01 Please see inline comments.

ViswanathanGanesh01 committed:
Model Clean Up

1. Updated the documentation code
2. Cleaned Buildings\Electrical\AC\ThreePhasesBalanced\Loads\MotorDrive\InductionMotors\BaseClasses

ViswanathanGanesh01 committed:
Merge branch 'issue2948_MotorDriver_update' into MBL---Electrical-Package-v1

JayHuLBL committed:
Merge pull request #4380 from ViswanathanGanesh01/MBL---Electrical-Package-v1

Model Clean Up

JayHuLBL committed:
deleted not needed files, fixed html errors, fixed file check errors [ci skip]

JayHuLBL committed:
removed BOM [ci shkip]

JayHuLBL committed:
changed the stop time [ci skip]

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/BaseClasses/SimVFD.mo:
```diff
@@ -0,0 +1,109 @@
+within Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.InductionMotors.BaseClasses;
+block SimVFD "Simple VFD maintaing constant V/F ratio"
```
Has this block been used by any other class? If not, can we delete it?

JayHuLBL committed:
clean up the InductionMotors package [ci skip]

JayHuLBL committed:
correcting formats [ci skip]

JayHuLBL committed:
Merge branch 'master' into issue2948_MotorDriver_update

JayHuLBL committed:
corrected format to align wth MBL conventions [ci skip]

JayHuLBL committed:
improved documentation [ci skip]

JayHuLBL committed:
changed controller type

ViswanathanGanesh01 committed:
format documentation

1. Added comments for instances
2. Added documentation for "Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/package.mo"
3. Removed option for having controller "Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/SquirrelCageDrive.mo" to normalize setPoi and mea.

JayHuLBL committed:
Merge pull request #4392 from ViswanathanGanesh01/MBL---Electrical-Package-v2

MBL Electrical Package

JayHuLBL committed:
corrected typo

JayHuLBL committed:
run unit test

JayHuLBL committed:
deleted not used class

JayHuLBL committed:
deleted self-connections

JayHuLBL committed:
propagating parameters [ci skip]

JayHuLBL committed:
propagated parameters

JayHuLBL committed:
proagated parameters

JayHuLBL committed:
added default names

JayHuLBL committed:
updated controllers

JayHuLBL committed:
Merge branch 'master' into issue2948_MotorDriver_update

JayHuLBL commented on Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/Coupled/Examples/Pump.mo:
```diff
@@ -0,0 +1,91 @@
+          -50,-40},{-50,21},{-80,21}}, color={0,127,255}));
+  connect(senMasFlo.m_flow,pum.m_flow)  annotation (Line(points={{-10,-29},{-10,
+          24},{-2,24}}, color={0,0,127}));
+  annotation (experiment(Tolerance=1e-6,StartTime=0,StopTime=200),
```
In general, I would suggest add more detailed introduction of each of the example models.

JayHuLBL reviewed (commented):
@ViswanathanGanesh01 See the inline comments.

JayHuLBL committed:
added missing comment

ViswanathanGanesh01 committed:
Merge branch 'lbl-srg:master' into MBL---Electrical-Package-v2

ViswanathanGanesh01 committed:
Updated the Documentation for all examples

ViswanathanGanesh01 committed:
Updated Validation Data Files

ViswanathanGanesh01 committed:
Updated Scripts Files

ViswanathanGanesh01 committed:
Updated Reference Results

ViswanathanGanesh01 committed:
Merge branch 'issue2948_MotorDriver_update' into MBL---Electrical-Package-v2

JayHuLBL committed:
Merge branch 'master' into issue2948_MotorDriver_update

JayHuLBL committed:
Merge pull request #4402 from ViswanathanGanesh01/MBL---Electrical-Package-v2

Updated Documentation MotorDrive Package

JayHuLBL committed:
cleaned the docuementation

JayHuLBL committed:
Merge branch 'master' into issue2948_MotorDriver_update

JayHuLBL committed:
merged master

JayHuLBL committed:
corrected typos

JayHuLBL committed:
fixed broken links

JayHuLBL commented:
@ViswanathanGanesh01 It will be ready for Michael to review after the CI tests passed.

JayHuLBL committed:
Merge branch 'master' into issue2948_MotorDriver_update

JayHuLBL committed:
Merge branch 'master' into issue2948_MotorDriver_update

JayHuLBL committed:
updated release note

JayHuLBL committed:
fixed typos

JayHuLBL committed:
fixed text format [ci skip]

ViswanathanGanesh01 committed:
Merge pull request #3 from lbl-srg/issue2948_MotorDriver_update

Merge Branch

ViswanathanGanesh01 committed:
Added Boolean Feature

Added Boolean Feature for the Models:
1. Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.Coupled.Chiller.mo
2. Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.Coupled.HeatPump.mo

Updated Examples:
1. Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.Coupled.Examples.Chiller.mo
2. Buildings.Electrical.AC.ThreePhasesBalanced.Loads.MotorDrive.Coupled.Examples.HeatPump.mo

ViswanathanGanesh01 committed:
updated the Example files

ViswanathanGanesh01 committed:
Revert "updated the Example files"

This reverts commit f4d242517a0f7fb2fc5ad77801705e0f472c1d34.

ViswanathanGanesh01 committed:
Addressed unnecessary changes of previous commit

JayHuLBL committed:
Merge pull request #4631 from ViswanathanGanesh01/MotorDrive_Coupled

Added Boolean Feature

JayHuLBL committed:
updated the example models

JayHuLBL committed:
deleted BOM

ViswanathanGanesh01 committed:
Merge pull request #4 from lbl-srg/issue2948_MotorDriver_update

Issue2948 motor driver update

JayHuLBL committed:
rerun unit tests

ViswanathanGanesh01 committed:
Merge 

Merge

ViswanathanGanesh01 committed:
Updated Models

Updated power calculation for boolean case.

ViswanathanGanesh01 committed:
Updated Reference Files

JayHuLBL committed:
Merge pull request #4635 from ViswanathanGanesh01/MotorDrive_Coupled

Boolean Feature MotorDrive.Coupled

JayHuLBL committed:
renamed varaibles

JayHuLBL committed:
renamed models

JayHuLBL committed:
corrected initial type
